core/src/main/java/org/geysermc/geyser/registry/populator/CustomItemRegistryPopulator.java

```diff
@@ -496,8 +509,13 @@ private static List<Float> toList(CustomRenderOffsets.OffsetXYZ xyz) {
         return List.of(xyz.x(), xyz.y(), xyz.z());
     }
 
+    @SuppressWarnings("unchecked")
     private static void setItemTag(NbtMapBuilder builder, String tag) {
-        builder.putList("item_tags", NbtType.STRING, List.of("minecraft:is_" + tag));
+        List<String> tagList = builder.get("item_tags") == null ? new ArrayList<>() : new ArrayList<>((List<String>) builder.get("item_tags"));
```

Comment:
We can probably use [computeIfAbsent](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#computeIfAbsent-K-java.util.function.Function-) here. 

All lists added to [NbtMapBuilder](https://github.com/CloudburstMC/NBT/blob/master/src/main/java/org/cloudburstmc/nbt/NbtMapBuilder.java)s are converted to [NbtList](https://github.com/CloudburstMC/NBT/blob/master/src/main/java/org/cloudburstmc/nbt/NbtList.java)s if they aren't one already, so we may as well avoid the extra copy. So, create a new NbtList as the `mappingFunction` (instead of an ArrayList). 

With that in mind, the `putList` call should be able to be removed.

Reply:
Inaccurate